### PR TITLE
Make ManagedCursorManager extensible

### LINKED ISSUE #169: Support wrapping on ManagedCursorManager
Currently the `NakadiClientBuilder` automagically wraps the `RequestFactory`, providing support for `problem` and adding authorization as defined by the `AccessTokenProvider`.

In case client wants to provide her own `CursorManager`, she has to implement this wrapping on their own as it seems, as the classes are not public.

### PR DESCRIPTION
**Pull request by jhorstmann**
By wrapping the ClientHttpRequestFactory with problem handling and authorization if needed

Fixes #169

### CODE REVIEW

File: fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java [hunk 1]
```diff
@@ -73,7 +74,15 @@ void setStreamId(String streamId) {
...
 
+    public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, AccessTokenProvider accessTokenProvider) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, accessTokenProvider), true);
+    }
+
     public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, null), true);
+    }
+
+    ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, boolean clientHttpRequestFactoryIsAlreadyWrapped) {
```
Comment (jhorstmann) [inline fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: Named constructors would be nice :)

Comment (lukasniemeier-zalando) [inline fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: Alternatively you could always wrap and change the wrapping code to something like
```
ClientHttpRequestFactory wrapIfNeeded(ClientHttpRequestFactory factory) {
  if (factory instanceof AuthorizedClientHttpRequestFactory) return factory;
  return wrapped(factory);
}
```

This way the wrapping decision would be at one place.

Comment (jhorstmann) [inline fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: I would have to check for both `instanceof` `AuthorizedClientHttpRequestFactory` and `ProblemHandlingClientHttpRequestFactory`, the authorization wrapping is only done if there is a ´AccessTokenProvider`.

**Comment by jhorstmann** [comment]
:+1:

**Pull request merged**